docker/index.ts

```diff
@@ -94,80 +97,83 @@ await $`
-    log.log("DEBUG", `Error was: ${err.stderr.toString()}, ${err.stdout.toString()}, path is: ${path.join(__dirname, gitPath)}`)
-    process.exit(1)
-})
-
-log.log("INFO", `Checking for changes in the repository: ${repo} on branch: ${branch} for user: ${user}`)
-log.log("DEBUG", `Also updating the repository upload url to avoid authentication issues`)
-//Switch to the correct branch
-await $`
-    cd ${gitPath}
-    git checkout ${branch}
-    #Update the push url to avoid authentication issues
-    git remote set-url --push origin https://${user}:${token}@github.com/${user}/${repo}
-`.quiet().catch((err)=>{
-    log.log("ERROR", `Failed to switch to branch: ${branch}, please check the branch and the repository: ${repo}`);
-    log.log("DEBUG", `Error was: ${err.stderr.toString()}`)
-    process.exit(1)
-})
-
-//Fetch the current commit hash
-let currentCommit:any = await $`
-    cd ${gitPath}
-    git rev-parse HEAD
-`.quiet().catch((err)=>{
-    log.log("ERROR", `Failed to fetch the current commit hash, please check the repository: ${repo}`);
-    log.log("DEBUG", `Error was: ${err.stderr.toString()}`)
-    process.exit(1)
-})
-currentCommit = currentCommit.stdout.toString().trim()
-log.log("INFO", `Current commit hash is: ${currentCommit}`)
-
-//Set the frequency to check for changes (Defaults to 1 Minute)
-let frequency = process.env.FREQUENCY ? process.env.FREQUENCY : 60000;
-log.log("DEBUG", `Checking for changes every ${frequency} milliseconds`)
-
-//Create the Octokit instance
-const octokit = new Octokit({
-    auth: process.env.GITHUB_TOKEN
-});
-
-const headers = await octokit.request('HEAD /')
-    .then((res)=>{
-        log.log("INFO", `Passed authentication Test with the GitHub API`)
-        return res.headers
-    })
-    .catch((err)=>{
-        log.log("ERROR", `Failed to authenticate with the GitHub API, please check your credentials`);
-        log.log("DEBUG", `Error was: ${err}`)
-        process.exit(1)
-    })
-if(!headers['x-oauth-scopes']){
-    log.log("ERROR", `The token does not have the required permissions to access the repository, please re-run with a token with the repo scope`);
-    process.exit(1)
-}
-const scopes = headers['x-oauth-scopes'].split(", ");
-if(!scopes.includes("repo")){
-    log.log("ERROR", `The token does not have the required permissions to access the repository, please re-run with a token with the repo scope`);
-    process.exit(1)
-}
-
-//set the timestamp of the next build that needs to be done
-let nextBuild = Date.now() + intervalValue;
-log.log("INFO", `Will build the next configuration at minimum: ${new Date(nextBuild).toISOString()}`)
-log.log("INFO", `Listening for changes in the repository: ${repo} on branch: ${branch} for user: ${user}`)
-let currentRun = 0;
-if(process.env.BUILD_ON_STARTUP && process.env.BUILD_ON_STARTUP == "true"){
-    log.log("WARN", "Will build the configuration on the first loop (so after the frequency you've set)")
+if (process.env.LOCAL_FLAKE != "true"){
```

Comment:
This should be turned into a constant at some point:
```suggestion
const isLocalFlake:boolean = process.env.LOCAL_FLAKE ? process.env.LOCAL_FLAKE == true : false
if (!isLocalFlake){
```